commands/fun/xkcd.js

```diff
@@ -18,17 +18,69 @@ module.exports = class XkcdCommand extends commando.Command {
 				default: false,
 			}],
 		});
+
+		this.XKCD_BASE_URL = "https://xkcd.com/";
+		this.XKCD_API_SUFFIX = "/info.0.json";
 	}
 
-	async run(msg, {id}) {
-		request({
-			uri: `http://wa.funsite.cz/xkcd/` + (id ? (id === 'latest' ? '?new' : '?id=' + id) : ''),
+	async apiRequest(location) {
+		return await request({
+			uri: this.XKCD_BASE_URL + location,
 			json: true,
 			headers: {
 				'User-Agent': 'WilsonBot',
 			},
-		}).then((a) => {
-			if (!a.image) return msg.say('Couldn\'t find that ID.');
+		});
+	}
+
+	parseComic(xkcdResponseJSON) {
+		let comic = {};
+		comic.id = xkcdResponseJSON.num;
+		comic.title = xkcdResponseJSON.safe_title;
+		comic.description = xkcdResponseJSON.alt;
+		comic.image = xkcdResponseJSON.img;
+		comic.url = this.XKCD_BASE_URL + xkcdResponseJSON.num;
+
+		return comic;
+	}
+
+	async fetchLatestComic() {
+		let responseJSON = await this.apiRequest(this.XKCD_API_SUFFIX);
+		return this.parseComic(responseJSON);
+	}
+
+	async fetchComicByID(id) {
+		let responseJSON;
+
+		try {
+			responseJSON = await this.apiRequest(id + this.XKCD_API_SUFFIX);
+		} catch(e) {
+			return {error: "Couldn't find that ID."};
+		}
+
+
+		return this.parseComic(responseJSON);
+	}
+
+	async fetchRandomComic() {
+		let latestComic = await this.fetchLatestComic();
+		let randomComicID = 1 + Math.floor(Math.random() * latestComic.id);
+		return await this.fetchComicByID(randomComicID);
+	}
+
+	fetchComic(arg) {
+		if (arg === "latest") {
+			return this.fetchLatestComic();
+		} else if (arg) {
+			return this.fetchComicByID(arg);
+		} else {
+			return this.fetchRandomComic();
+		}
+	}
+
+	async run(msg, {arg}) {
```

Comment:
The command argument is defined with key `id` on line 15, therefore this destructuring results in a value of `undefined` for `arg`. It should either also be called `id` here or the argument key could be changed to `arg` (the former seems like a nicer option imo).